pkg/validate/scoped/hydrate/namespace_hydrator.go

```diff
@@ -143,7 +143,7 @@ func buildSelectorMap(ctx context.Context, c client.Client, objs *objects.Scoped
 		var selectedNamespaces []string
 		// Select dynamic/on-cluster Namespaces that match the NamespaceSelector's labels.
 		for _, ns := range nsList.Items {
-			if selector.Matches(labels.Set(ns.GetLabels())) {
+			if ns.Status.Phase != corev1.NamespaceTerminating && selector.Matches(labels.Set(ns.GetLabels())) {
```

Comment:
Is there a case when these were previously selected, but now the namespace is being deleted that would cause the objects to be removed from the UI & ResourceGroup before they were deleted?

Doesn't this cause the same side effect as https://github.com/GoogleContainerTools/kpt-config-sync/pull/1191?

If so, I think it'd make more sense to include this part of the change in that PR so that those namespace deletion changes are all together. Otherwise, if we need to revert them later it would also revert the unrelated nsControllerState fix in this PR.

Reply:
> Is there a case when these were previously selected, but now the namespace is being deleted that would cause the objects to be removed from the UI & ResourceGroup before they were deleted?

The Namespace finalizer should delete the previously selected resources. It is an improvement because the resource should not be selected before, and would never become managed by Config Sync (i.e. show in the UI & ResourceGroup). Hence, we should not care if they're deleted or not. It is the terminating Namespace finalizer's job.

> Doesn't this cause the same side effect as https://github.com/GoogleContainerTools/kpt-config-sync/pull/1191?

It is different. #1191 aims to unblock RepoSync deletion, while this change applies to managed resources. Since the resources are dynamically selected by the NamespaceSelector, it should be okay to unselect them when the Namespace is terminating.
I don't think the improvement would break anything.